src/process.cr

```diff
@@ -1,60 +1,47 @@
-require "c/signal"
-require "c/stdlib"
-require "c/sys/resource"
-require "c/unistd"
+require "crystal/system/process"
 
 class Process
   # Terminate the current process immediately. All open files, pipes and sockets
   # are flushed and closed, all child processes are inherited by PID 1. This does
   # not run any handlers registered with `at_exit`, use `::exit` for that.
   #
   # *status* is the exit status of the current process.
-  def self.exit(status = 0)
-    LibC.exit(status)
+  def self.exit(status = 0) : NoReturn
+    exit_system(status)
   end
 
   # Returns the process identifier of the current process.
-  def self.pid : LibC::PidT
-    LibC.getpid
+  def self.pid : Int64
+    pid_system
   end
 
   # Returns the process group identifier of the current process.
-  def self.pgid : LibC::PidT
+  def self.pgid : Int64
     pgid(0)
   end
 
   # Returns the process group identifier of the process identified by *pid*.
-  def self.pgid(pid : Int32) : LibC::PidT
-    ret = LibC.getpgid(pid)
-    raise Errno.new("getpgid") if ret < 0
-    ret
+  def self.pgid(pid : Int64) : Int64
+    Process.pgid_system(pid)
   end
 
   # Returns the process identifier of the parent process of the current process.
-  def self.ppid : LibC::PidT
-    LibC.getppid
+  def self.ppid : Int64
+    ppid_system
   end
 
   # Sends a *signal* to the processes identified by the given *pids*.
-  def self.kill(signal : Signal, *pids : Int)
+  def self.kill(signal : Signal, *pids : Int64) : Nil
     pids.each do |pid|
-      ret = LibC.kill(pid, signal.value)
-      raise Errno.new("kill") if ret < 0
+      Process.new(pid).kill(signal)
```

Comment:
This isn't good on unix since you need to allocate a `Process` and call `waitpid` just to send a signal. I think there needs to be both` .signal_system(signal, pid)` and `#signal_system(signal)`.

Reply:
`signal_system` is internal API, having a class method would be enough. It should be used for the (system specific) implementations of both class and instance methods `kill`, `terminate` and `interrupt`, respectively.
Adding an instance method `#signal_system(signal)` wouldn't be super useful when you can just call `Process.signal_system(signal, @pid)` instead. It would only be internal API anyway.

Reply:
@straight-shoota that's fine on unix, but on windows you call `TerminateProcess` on the handle.

Converting from the process handle to the PID just to convert to the process handle is inefficient, and that optimization is why there is both a class and an instance method: the `Process` already has the handle open.

Reply:
Oh yes, you're right.